instrumentation/github.com/gin-gonic/gin/otelgin/gintrace.go

```diff
@@ -86,7 +86,18 @@ func Middleware(service string, opts ...Option) gin.HandlerFunc {
 			opts = append(opts, oteltrace.WithAttributes(rAttr))
 		}
 		ctx, span := tracer.Start(ctx, spanName, opts...)
-		defer span.End()
+		defer func() {
+			if cfg.RecordPanicConfig.enabled {
+				if r := recover(); r != nil {
+					err := fmt.Errorf("error handling request: %s", r)
```

Comment:
What is the added value of creating a new error here over recording the original one?

If we do want to create a new error, we should wrap the original one:
```suggestion
					err := fmt.Errorf("error handling request: %w", r)
```

Reply:
i'll update to record the original error